### PR TITLE
[Multimat 1/n] Slam interface changes for Multimat

### PR DESCRIPTION
**Pull request by kanye-quest**
# Summary

- This PR is a cherry-picking of the Slam-specific changes from #565, synchronized with develop.
- Some ImplicitGrid-related fixes have been added, including a `CoreArrayIndirection` indirection policy and an allocator-aware `IndirectionPolicy::create()`

### CODE REVIEW

File: src/axom/spin/ImplicitGrid.hpp [hunk 1]
```diff
@@ -71,10 +71,10 @@ class ImplicitGrid
   using BinSet = slam::OrderedSet<IndexType, IndexType, SizePolicy>;
 
   using BitsetType = slam::BitSet;
-  using BinBitMap = slam::Map<slam::Set<IndexType, IndexType>,
-                              BitsetType,
-                              slam::policies::StrideOne<IndexType>,
-                              slam::policies::ArrayStorage<BitsetType>>;
+  using BinBitMap = slam::Map<BitsetType,
+                              slam::Set<IndexType, IndexType>,
+                              slam::policies::CoreArrayIndirection<IndexType, BitsetType>,
```
Comment (kennyweiss) [inline src/axom/spin/ImplicitGrid.hpp]: Neat!

**Review by kennyweiss — APPROVED** [review]
Thanks @kanye-quest !

* Are there any slam-related changes in #565 that are not included here?
* Are you planning to update #565 after this is merged?

**Comment by kanye-quest** [comment]
@kennyweiss -- I think this should encompass all the changes to Slam from the multimat PR. After this is merged, we should be able to somewhat-cleanly update the multimat PR with develop.

File: src/axom/slam/tests/slam_set_BivariateSet.cpp [hunk 2]
```diff
@@ -0,0 +1,467 @@
...
+  {
+    BSet* bset = new slam::ProductSet<S1, S2>(this->m_set1, this->m_set2);
+    EXPECT_TRUE(bset->isValid(true));
+    delete bset;
+  }
+}
+
+//-----------------------------------------------------------------------------
+
+// Tests BivariateSeet::firstSetSize(), secondSetSize()
```
Comment (agcapps) [inline src/axom/slam/tests/slam_set_BivariateSet.cpp:346]:
```suggestion
// Tests BivariateSet::firstSetSize(), secondSetSize()
```
Check for copy/paste of this typo elsewhere

File: src/axom/slam/BivariateSet.hpp [hunk 3]
```diff
@@ -43,9 +48,10 @@ namespace slam
  *
  *  For example, a 2 x 4 sparse matrix below:
  *     \code
- *        0  1  2  3
- *     0  a     b
- *     1     c     d
+ *         0  1  2  3
+ *         _  _  _  _
+ *     0 | a     b
+ *     1 |    c     d
```
Comment (agcapps) [inline src/axom/slam/BivariateSet.hpp:54]: Thanks!  this is an improvement.

File: src/axom/slam/BivariateSet.hpp [hunk 4]
```diff
@@ -139,6 +159,13 @@ class BivariateSet
    */
   virtual PositionType findElementFlatIndex(PositionType pos1) const = 0;
 
+  /**
+   * \brief Finds the range of indices o
```
Comment (agcapps) [inline src/axom/slam/BivariateSet.hpp:163]: Did you stop typing in mid-word?

**Pull request merged**